### PR TITLE
Add support for checking equality modulo NaN

### PR DESCRIPTION
**Pull request by jystic**
This adds a new type class `NanEq` which is the same as `Eq` but where `nanEq NaN NaN = True`.

I have changed the failing test https://github.com/ambiata/icicle/issues/536 to use the new `=~=` operator, but I am sure there will be other failures in time as I've added `DoubleT` back in to the allowed types which can be generated.

The plan would be to replace `===` with `=~=` whenever we come across a failure due to `NaN`.

This fixes https://github.com/ambiata/icicle/issues/385 and https://github.com/ambiata/icicle/issues/536.

! @amosr @tranma

### CODE REVIEW

**Comment by amosr** [comment]
wow! 👍

**Comment by tranma** [comment]
Javascripty
👍

File: icicle-source/src/Icicle/Encoding.hs [hunk 1]
```diff
@@ -375,74 +376,73 @@ sourceTypeOfEncoding e
...
+renderJsonValue :: Value -> Maybe Text
+renderJsonValue = \case
+   Tombstone ->
+     Nothing
+   StringValue v ->
+     pure $ renderJsonString v
+   IntValue v ->
+     pure . T.pack $ show v
+   DoubleValue v ->
+     pure . T.decodeUtf8 $ renderDouble v
```
Comment (jystic) [inline icicle-source/src/Icicle/Encoding.hs]: We do the JSON rendering ourself here instead of using Aeson, so that we can use `renderDouble` from Anemone

**Comment by jystic** [comment]
On top of https://github.com/ambiata/anemone/pull/53

**Pull request merged**